pkg/proxy/iptables/proxier.go

```diff
@@ -656,6 +664,36 @@ func (proxier *Proxier) OnEndpointsUpdate(allEndpoints []api.Endpoints) {
 	proxier.deleteEndpointConnections(staleConnections)
 }
 
+// needToUpdate returns if the input endpoints is different from the last batch
+// This is calculated by comparing the checksum of EndpointSubsets.
+func (proxier *Proxier) needToUpdate(allEndpoints []api.Endpoints) bool {
+	checksumMap := make(map[string]uint32)
+	hash := fnv.New32a()
```

Comment:
I'm no expert on the fnv family of hash functions but the implementation seems to mutate the value such that one probably needs to invoke reset: https://golang.org/src/hash/fnv/fnv.go (though I'm assuming you verified it, just checking)